File: mobile/ios/LoHiTrade/LoHiTrade/Models/ChatModels.swift
import Foundation

// MARK: - Chat Message (Req 18.1)

/// Role of a chat message participant.
enum ChatRole: String, Codable, Equatable {
    case user
    case assistant
}

/// A single message in a chatbot conversation.
struct ChatMessage: Codable, Identifiable, Equatable {
    let id: String
    let role: ChatRole
    let content: String
    let chartImageURL: String?
    let timestamp: String

    enum CodingKeys: String, CodingKey {
        case id, role, content
        case chartImageURL = "chart_image_url"
        case timestamp
    }

    /// Convenience URL for the chart image, if present.
    var chartURL: URL? {
        guard let chartImageURL else { return nil }
        return URL(string: chartImageURL)
    }
}

// MARK: - Chat Session

/// Represents a chatbot conversation session.
struct ChatSession: Codable, Identifiable, Equatable {
    let id: String
    let messages: [ChatMessage]
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id, messages
        case createdAt = "created_at"
    }
}

// MARK: - API Request / Response

/// Request body for POST /chatbot/message.
struct ChatMessageRequest: Encodable {
    let message: String
}

/// Response from POST /chatbot/message.
struct ChatMessageResponse: Decodable {
    let message: ChatMessage
}

/// Response from GET /chatbot/history.
struct ChatHistoryResponse: Decodable {
    let messages: [ChatMessage]
}

/// Response from DELETE /chatbot/session.
struct ChatSessionDeleteResponse: Decodable {
    let success: Bool
    let message: String
}

File: mobile/android/LoHiTrade/app/src/main/java/com/lohitrade/data/models/ChatModels.kt
package com.lohitrade.data.models

import com.google.gson.annotations.SerializedName

/**
 * Data models for the AI chatbot (Req 18.1, 20.7).
 */

/** Role of a chat message participant. */
enum class ChatRole {
    @SerializedName("user") USER,
    @SerializedName("assistant") ASSISTANT
}

/** A single chat message with optional chart image URL. */
data class ChatMessage(
    val id: String,
    val role: ChatRole,
    val content: String,
    @SerializedName("chart_image_url") val chartImageUrl: String? = null,
    val timestamp: String
)

/** Request body for POST /chatbot/message. */
data class ChatMessageRequest(
    val message: String
)

/** Response from POST /chatbot/message. */
data class ChatMessageResponse(
    val message: ChatMessage
)

/** Response from GET /chatbot/history. */
data class ChatHistoryResponse(
    val messages: List<ChatMessage>
)

/** Response from DELETE /chatbot/session. */
data class ChatSessionDeleteResponse(
    val success: Boolean,
    val message: String
)
